custom/icds_reports/utils/aggregation_helpers/distributed/inactive_awws.py

```diff
@@ -45,7 +45,7 @@ def data_from_ucr_query(self):
                 FIRST_VALUE(form_date) OVER forms as first_submission,
                 LAST_VALUE(form_date) OVER forms as last_submission
             FROM "{ucr_tablename}"
-            WHERE inserted_at >= %(last_sync)s AND month > %(six_months_ago)s AND form_date <= %(now)s
+            WHERE inserted_at >= %(last_sync)s AND month > %(one_month_ago)s AND form_date <= %(now)s
```

Comment:
@calellowitz  @emord Does the 6 months come from the fact that, we would consider 6 months of delayed data? If so, then this change would break that. (The `form_date` comes from `form.meta.timeEnd`). Or is this fine?

Reply:
This query might be fast enough if we added an index to `inserted_at`. From my understanding we do have a long tail of late submissions which is why we run the aggregation of the previous month.

Reply:
Okay, I think it will be large index given that this table will have all forms, I think that should be okay? Separately, does it still make sense to make this change of 6 months to 1 month?

Reply:
Or why can't the `received_on` be use instead of `form_date` (`form.meta.timeEnd`)? That would make it possible to consider only few days worth of forms and still not miss the tail submissions that come in late.

Reply:
> Separately, does it still make sense to make this change of 6 months to 1 month?

I think that's the 6 months is intended to process late submissions which certainly do occur past one month. 

> Or why can't the received_on be use instead of form_date
> That would make it possible to consider only few days worth of forms

I think that's what the filter on inserted_at was intendend to do.

Reply:
Okay, I will close this and make another PR to add an index on `inserted_at`

Reply:
Okay, makes sense then to add index to inserted_at. `inserted_at >= %(last_sync)s ` is sufficient to pull the latest updates (including) the tail, so it's not very clear why the other two filters on month and form_date are required (they are not partitioned anyway). 

Adding index on `inserted_at` is a bit tricky, since that column isn't defined in the UCR spec, but is added automatically by the UCR framework. Is there a way to add index to this column using the UCR framework? Otherwise the only choice is to create the index manually on the table (but it's not good, since the migration won't exist as a code)

Reply:
> so it's not very clear why the other two filters on month and form_date are required

I think filter on month is to limit the tail to just 6 month not older than that filter on form date could be because of the occurrence of forms came with `timeend` value in future.

Reply:
@sravfeyn We have other data sources that index on `inserted_at` such as https://github.com/dimagi/commcare-hq/blob/7100e7db6cf19653525dd7fde968a75071768fa8/custom/icds_reports/ucr/data_sources/ccs_record_cases_monthly_v2.json#L2770-L2774

Before deploying that, you should create it on the table concurrently with something like `CREATE INDEX CONCURRENTLY usage_forms_inserted_at_idx ON usage_forms_table (inserted_at)` so that it is not created in a blocking manner when added to the data source. (this was documented https://confluence.dimagi.com/display/commcarehq/Using+pghero but I suppose we haven't used it recently)